### PR TITLE
docs: Document Isso Project Infrastructure

### PR DESCRIPTION
**Pull request by ix5**
## Checklist
- [x] All new and existing **tests are passing**
- [ ] (If adding features:) I have added tests to cover my changes
- [x] (If docs changes needed:) I have updated the **documentation** accordingly.
- [ ] I have added an entry to `CHANGES.rst` because this is a user-facing change or an important bugfix
- [x] I have written **proper commit message(s)**
      

## What changes does this Pull Request introduce?
Try to codify the new reality of the project since Martin stepped down and make sense of all the informal or mostly-one-off processes we are currently running.

I've also created [isso-demo-config](https://github.com/isso-comments/isso-demo-config) and [isso-marketing-material](https://github.com/isso-comments/isso-marketing-material) as well as linked [ansible-role-isso](https://git.ix5.org/felix/ansible-role-isso) to coincide with this PR.

## Why is this necessary?
Brought about by recent events, I thought about the nature of maintainership a bit and realized that we have a) a pretty significant bus factor and b) our infra could be a bit opaque to newcomers. Collecting all the information and writing it down in one place also might help highlight a few blind spots and areas for improvement.

### CODE REVIEW

**Review by jelmer — COMMENTED** [review]
Deleted

File: docs/docs/contributing/infrastructure.rst [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+.. _PyPI: https://pypi.org/project/isso/
+
+PyPI (Python Package Index)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Release rights for the `isso PyPI`_ project are held by @jelmer, @posativ, @blatinier and @ix5.
+
+Since `PGP support was dropped from PyPI`_ in 2023, PGP keys for signing the
+package are no longer relevant.
+
```
Comment (jelmer) [inline docs/docs/contributing/infrastructure.rst:91]: This seems irrelevant here.

File: docs/docs/contributing/infrastructure.rst [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+``Inherit access from source repository`` turned on).
+
+The `ghcr.io/isso-comments/isso-js-testbed`_ image for running ``Jest``-based
+unit and integration tests is built and pushed manually by @ix5 so far.
+
+.. _ghcr.io/isso-comments/isso docker image: https://github.com/isso-comments/isso/pkgs/container/isso
+.. _ghcr.io/isso-comments/isso-js-testbed: https://github.com/orgs/isso-comments/packages/container/package/isso-js-testbed
+
+Arch User Repository (AUR)
+^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (jelmer) [inline docs/docs/contributing/infrastructure.rst:110]: This overlaps with our packaging documentation, and isn't part of the upstream project or it's infrastructure.

**Comment by posativ** [comment]
LGTM, I just checked PyPi and @blatinier has the Maintainer role, do we need to change this? Is there any point to making an organization on PyPi?

**Review by posativ — APPROVED** [review]

**Comment by jelmer** [comment]
> LGTM, I just checked PyPi and @blatinier has the Maintainer role, do we need to change this? Is there any point to making an organization on PyPi?

I think organizations are still an experimental feature that needs to be requested and are hard to get.

File: docs/docs/contributing/infrastructure.rst (hunk 1 above)
Comment (ix5) [reply docs/docs/contributing/infrastructure.rst:91]: My intention was to convey that there are no "release keys" _of significance to PyPI_ being held by anyone, but I agree with you, that's history and not currently relevant.

File: docs/docs/contributing/infrastructure.rst (hunk 2 above)
Comment (ix5) [reply docs/docs/contributing/infrastructure.rst:110]: Agreed.

I'll document that in `Prebuilt Packages` in [docs/docs/reference/installation.rst](docs/docs/reference/installation.rst) in a different PR, along with documenting that the Debian package is gone.

**Comment by ix5** [comment]
> LGTM, I just checked PyPi and @blatinier has the Maintainer role, do we need to change this? Is there any point to making an organization on PyPi?

Mainainer role is sufficient for releases afaics. Also seems that Benoit is no longer much (if at all) involved with the project.
 
> I think organizations are still an experimental feature that needs to be requested and are hard to get.

Yes, also I don't immediately see the added benefit of that feature for us.

We might investigate using the "Publish from GitHub" feature ([direct link: "Trusted Publisher Management" for isso](https://pypi.org/manage/project/isso/settings/publishing/)). That might simplify releasing new versions further, since we already publish docker images on any `git tag`.

>  OpenID Connect (OIDC) provides a flexible, credential-free mechanism for delegating publishing authority for a PyPI package to a trusted third party service, like GitHub Actions.
>
> PyPI users and projects can use trusted publishers to automate their release processes, without needing to use API tokens or passwords.
